test/conformance/configuration_test.go

```diff
@@ -0,0 +1,139 @@
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"github.com/knative/serving/test"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"reflect"
+	"testing"
+)
+
+func TestUpdateConfigurationMetadata(t *testing.T) {
+	clients := setup(t)
+
+	logger := logging.GetContextLogger("TestUpdateConfigurationMetadata")
+
+	var names test.ResourceNames
+	names.Service = test.AppendRandomString("pizzaplanet-service", logger)
+	names.Config = names.Service
+
+	defer tearDown(clients, names)
+	test.CleanupOnInterrupt(func() { tearDown(clients, names) }, logger)
+
+	logger.Infof("Creating new configuration %s", names.Config)
+	err := test.CreateConfiguration(logger, clients, names, test.ImagePath(image1), &test.Options{})
+	if err != nil {
+		t.Fatalf("Failed to create configuration %s", names.Config)
+	}
+
+	cfg := reloadConfiguration(names.Config, clients, t)
+
+	logger.Info("The Configuration will be updated with the name of the Revision once it is created")
+	names.Revision, err = waitForConfigurationLatestCreatedRevision(clients, names)
+	if err != nil {
+		t.Fatalf("Configuration %s was not updated with the new revision: %v", names.Config, err)
+	}
+
+	cfg = reloadConfiguration(names.Config, clients, t)
+
+	logger.Infof("Updating labels of Configuration %s", names.Config)
+	cfg.Labels =  map[string]string{
+		"labelX": "abc",
+		"labelY": "def",
+	}
+	cfg, err = clients.ServingClient.Configs.Update(cfg)
+	if err != nil {
+		t.Fatalf("Failed to update labels for Configuration %s: %v", names.Config, err)
+	}
+
+	if err = waitForConfigurationLabelsUpdate(clients, names, cfg.Labels); err != nil {
+		t.Fatalf("The labels for Configuration %s were not updated: %v", names.Config, err)
+	}
+
+	cfg = reloadConfiguration(names.Config, clients, t)
+	expected := names.Revision
+	actual := cfg.Status.LatestCreatedRevisionName
+	if expected != actual {
+		t.Errorf("Did not expect a new Revision after updating labels for Configuration %s - expected Revision: %s, actual Revision: %s",
+			names.Config, expected, actual)
+	}
+
+	logger.Infof("Updating annotations of Configuration %s", names.Config)
+	cfg.Annotations = map[string]string{
+		"annotationA": "123",
+		"annotationB": "456",
+	}
+	cfg, err = clients.ServingClient.Configs.Update(cfg)
+	if err != nil {
+		t.Fatalf("Failed to update annotations for Configuration %s: %v", names.Config, err)
+	}
+
+	if err = waitForConfigurationAnnotationsUpdate(clients, names, cfg.Annotations); err != nil {
+		t.Fatalf("The annotations for Configuration %s were not updated: %v", names.Config, err)
+	}
+
+	cfg = reloadConfiguration(names.Config, clients, t)
+	actual = cfg.Status.LatestCreatedRevisionName
+	if expected != actual {
+		t.Errorf("Did not expect a new Revision after updating annotations for Configuration %s - expected Revision: %s, actual Revision: %s",
+			names.Config, expected, actual)
+	}
+}
+
+func reloadConfiguration(name string, clients *test.Clients, t *testing.T) *v1alpha1.Configuration {
```

Comment:
The word `reload` has thrown me off course here when reviewing, does it make sense to name the function `fetchConfiguration`?

Reply:
Agreed.